libs/common/src/itt.hpp

```diff
@@ -0,0 +1,110 @@
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+/**
+ * @brief Defines API to trace using Intel ITT.
+ * @file itt.hpp
+ */
+
+#pragma once
+#include "itt_domains.hpp"
+#include <string>
+
+namespace openvino
+{
+    namespace itt
+    {
+        typedef struct domain_ {} *domain_t;
+        typedef struct handle_ {} *handle_t;
+
+        namespace internal
+        {
+            domain_t domain(char const* name);
+            handle_t handle(char const* name);
+            void taskBegin(domain_t d, handle_t t);
+            void taskEnd(domain_t d);
+            void threadName(const char* name);
+        }
+
+        inline void threadName(const char* name)
+        {
+            internal::threadName(name);
+        }
+
+        inline void threadName(const std::string &name)
+        {
+            internal::threadName(name.c_str());
+        }
+
+        inline handle_t handle(char const *name)
+        {
+            return internal::handle(name);
+        }
+
+        inline handle_t handle(const std::string &name)
+        {
+            return internal::handle(name.c_str());
+        }
+
+        template <typename Tag>
+        domain_t domain()
+        {
+            static auto d = internal::domain(Tag::name());
+            return d;
+        }
+
+        template <typename Tag>
+        handle_t handle()
+        {
+            static auto h = internal::handle(Tag::name());
+            return h;
+        }
+
+        template <typename DomainTag, typename TaskTag = void>
+        struct ScopedTask
+        {
+            ScopedTask() noexcept
+            {
+                internal::taskBegin(domain<DomainTag>(), handle<TaskTag>());
+            }
+            ~ScopedTask() noexcept { internal::taskEnd(domain<DomainTag>()); }
+            ScopedTask(const ScopedTask&) = delete;
+            ScopedTask& operator=(const ScopedTask&) = delete;
+        };
+
+        template <typename DomainTag>
+        struct ScopedTask<DomainTag, void>
+        {
+            ScopedTask(handle_t taskHandle) noexcept
+            {
+                internal::taskBegin(domain<DomainTag>(), taskHandle);
+            }
+            ~ScopedTask() noexcept { internal::taskEnd(domain<DomainTag>()); }
+            ScopedTask(const ScopedTask&) = delete;
+            ScopedTask& operator=(const ScopedTask&) = delete;
+        };
+
+#define OV_SCOPED_TASK(domainTag, taskTag, taskName)                                               \
+    struct Task##taskTag                                                                           \
```

Comment:
May be use __LINE__ macro

Reply:
Yes, I can use the __LINE__ macro, but it can lead to type clashes.

Reply:
Yes it's good idea.